Open source build improvements

jasonmalinowski opened the pull request:
Make several improvements to Jenkins and community building:

1) If we're running on a non-internal-developer machine, just don't try to deploy extensions since that'll always fail
2) Remove the CIBuild flag entirely -- community and Jenkins builds should just do the same thing
3) Run the same tests everywhere

Reviewers: @jaredpar, @amcasey, @davkean, @tannergooding, @agocke

jaredpar commented:
:+1: 

Build seems to have failed with the following:

``` 
"d:\j\workspace\dotnet_roslyn_windows_prtest\src\Samples\..\..\packages\NuGet.CommandLine.2.8.5\tools\NuGet.exe pack Diagnostic.nuspec -NoPackageAnalysis -OutputDirectory d:\j\workspace\dotnet_roslyn_windows_prtest\Binaries\Debug\" exited with code 3.
  MakeConstCS.UnitTests -> 
````

Looks unrelated.  Will retest.

jaredpar commented:
@dotnet-bot  test this please

jaredpar commented:
Hmm, failure looks real.

tannergooding commented:
@dotnet-bot test this please

Going to try and repro it on the build machine.

tannergooding commented:
@dotnet-bot test this please

jaredpar commented:
@tannergooding the problem is this line in the build file:

> https://github.com/dotnet/roslyn/blob/master/src/Samples/CSharp/AsyncPackage/AsyncPackage.csproj#L194

@jasonmalinowski removed the `CIBuild` option hence this is getting triggered on the build server where it can't run correctly.

jasonmalinowski commented:
I removed the check in AsyncPackage.csproj because once I cleaned up everything else that was the *only* remaining thing depending upon CIBuild. I'll investigate further what's up.

tmeschter commented:
@jasonmalinowski I already have a pull request open to remove AsyncPackage entirely, as I've already moved to the dotnet/roslyn-analyzers.

jasonmalinowski commented:
@tmeschter, when do you anticipate that'll get merged in?

amcasey commented on src/Tools/Source/RunTests/TestRunner.cs:
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
-                errorOutput = processOutput.ErrorLines.Any()
-                    ? processOutput.ErrorLines.Aggregate((x, y) => x + Environment.NewLine + y)
-                    : string.Format("xunit produced no error output but had exit code {0}", processOutput.ExitCode);
+                errorOutput.AppendLine($"Command: {_xunitConsolePath} {builder.ToString()}");
```
Isn't ```.ToString()``` implied?

amcasey commented on src/Tools/Source/RunTests/TestRunner.cs:
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
-                errorOutput = string.Format("Command: {0} {1}", _xunitConsolePath, builder.ToString())
-                    + Environment.NewLine
-                    + errorOutput;
+                if (processOutput.ErrorLines.Any())
```
Extract local for ```processOutput.ErrorLines```?

amcasey commented on src/Tools/Source/RunTests/TestRunner.cs:
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
+                }
+                else
+                {
+                    errorOutput.Append($"xunit produced no error output but had exit code {processOutput.ExitCode}");
```
```AppendLine```?

amcasey commented on src/Tools/Source/RunTests/TestRunner.cs:
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
                 }
 
-                errorOutput = processOutput.ErrorLines.Any()
-                    ? processOutput.ErrorLines.Aggregate((x, y) => x + Environment.NewLine + y)
```
This is hilarious.

amcasey commented:
LGTM, assuming you tested the three cases (Roslyn dev, external dev, Jenkins).

jasonmalinowski commented:
Jenkins will be fixed once @tmecshter deletes the thing causing me pain. I've verified BuildAndTest works on a clean box without dogfood tools (although I needed to disable strong name verification to get around the test resources problem). Once those are both fixed I'll also do the internal dev test too.

jasonmalinowski commented:
And @tmeschter as removed the async analyzers, so I've rebased and we'll see what Jenkins thinks now.

jaredpar commented:
@dotnet-bot test this please 

One more Jenkins run as we changed MSBuild cross plat at the same time.
